internal/helmdeployer/install.go

```diff
@@ -119,89 +137,219 @@ func (h *Helm) install(ctx context.Context, bundleID string, manifest *manifest.
+	namespace string,
+	timeout time.Duration,
+	options fleet.BundleDeploymentOptions,
+	pr *postRender,
+	dryRunCfg dryRunConfig,
+) (*releasev1.Release, error) {
+	logger := log.FromContext(ctx)
+	u := action.NewInstall(cfg)
+
+	h.configureInstallAction(u, cfg, releaseName, namespace, timeout, options, pr, dryRunCfg)
+
+	if !dryRunCfg.DryRun {
+		logger.Info("Installing helm release")
+	}
+
+	rel, err := u.Run(chart, values)
+	if err != nil {
+		return nil, err
+	}
+
+	return assertRelease(rel)
+}
+
+// configureInstallAction configures a Helm Install action with Fleet-specific options,
+// including timeout, wait strategies, and server-side apply settings.
+func (h *Helm) configureInstallAction(u *action.Install, cfg *action.Configuration, releaseName, namespace string, timeout time.Duration, options fleet.BundleDeploymentOptions, pr *postRender, dryRunCfg dryRunConfig) {
+	if cfg.Capabilities != nil {
+		if cfg.Capabilities.KubeVersion.Version != "" {
+			u.KubeVersion = &cfg.Capabilities.KubeVersion
 		}
-		u.TakeOwnership = options.Helm.TakeOwnership
-		u.EnableDNS = !options.Helm.DisableDNS
-		u.Replace = true
-		u.Atomic = options.Helm.Atomic
-		u.ReleaseName = releaseName
-		u.CreateNamespace = true
-		u.Namespace = defaultNamespace
-		u.Timeout = timeout
-		u.DryRun = dryRunCfg.DryRun
-		u.DryRunOption = dryRunCfg.DryRunOption
-		u.SkipSchemaValidation = options.Helm.SkipSchemaValidation
-		u.PostRenderer = pr
-		u.WaitForJobs = options.Helm.WaitForJobs
-		if u.Timeout > 0 {
-			u.Wait = true
+		if cfg.Capabilities.APIVersions != nil {
+			u.APIVersions = cfg.Capabilities.APIVersions
 		}
-		if !dryRunCfg.DryRun {
-			logger.Info("Installing helm release")
+	}
+	u.TakeOwnership = options.Helm.TakeOwnership
+	// Disable server-side apply when taking ownership to avoid managedFields validation errors.
+	// When adopting existing resources, they have managedFields populated by Kubernetes,
+	// but server-side apply requires managedFields to be nil. Using client-side apply (three-way merge) instead.
+	if u.TakeOwnership {
+		u.ServerSideApply = false
+	}
+	u.EnableDNS = !options.Helm.DisableDNS
+	u.Replace = true
+	u.RollbackOnFailure = options.Helm.Atomic
+	u.ReleaseName = releaseName
+	u.CreateNamespace = true
+	u.Namespace = namespace
+	u.Timeout = timeout
+	// Configure dry-run strategy based on template mode and dryRunConfig.
+	// Template mode (h.template) requires DryRunClient to render without cluster interaction.
+	// If DryRunOption is "server", use DryRunServer to allow lookup functions to query the cluster.
+	// Otherwise, use DryRunClient for client-only dry run or DryRunNone for actual execution.
+	if h.template {
+		u.DryRunStrategy = action.DryRunClient
+	} else if dryRunCfg.DryRun {
+		if dryRunCfg.DryRunOption == "server" {
+			u.DryRunStrategy = action.DryRunServer
+		} else {
+			u.DryRunStrategy = action.DryRunClient
 		}
-		return u.Run(chart, values)
+	} else {
+		u.DryRunStrategy = action.DryRunNone
+	}
```

Comment:
nit: since this works the same way in both install and upgrade cases, wouldn't it make sense to isolate this logic into a separate function?